File: Data Structures & Algorithms/valid-binary-search-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public boolean isValidBST(TreeNode root) {
        if(root == null){
            return true; 
        }

        TreeNode left = root.left; 
        TreeNode right = root.right;

        if(left != null && left.val >= root.val){
            return false; 
        } 

        if(right != null && right.val =< root.val){
            return false;
        }

        return isValidBST(left) && isValidBST(right); 
  }
}

File: Data Structures & Algorithms/balanced-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right


class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        if(not root): 
            return True
        
        def dfs(curr):

            if(not curr): 
                return 0 
            
            left = dfs(curr) + 1 
            right = dfs(curr) + 1

            height = max(left, right)

            return height

        leftHeight = dfs(curr.left)
        rightHeight = dfs(curr.right)

        if(abs(leftHeight - rightHeight) > 1): 
            return False

        return True